src/frontend/src/optimizer/rule/distinct_agg.rs

```diff
@@ -66,37 +55,68 @@ impl DistinctAggRule {
         Box::new(DistinctAggRule {})
     }
 
+    /// Construct `Expand` for distinct aggregates.
+    ///
+    /// To simplify, we will first deduplicate `column_subsets` and then skip building
+    /// `Expand` if there is only one `subset`.
     fn build_expand(
         input: PlanRef,
-        group_keys: &[usize],
-        distinct_aggs: &[PlanAggCall],
-        non_distinct_aggs: &[PlanAggCall],
-    ) -> PlanRef {
+        group_keys: &mut Vec<usize>,
+        agg_calls: &mut Vec<PlanAggCall>,
+    ) -> Option<(PlanRef, Vec<usize>, bool)> {
```

Comment:
It is better to include the following in the doc comment:
* What is each returned value
* What mutations are done to parameters declared with `&mut`